skins/oasis/modules/UserPagesHeaderController.class.php

```diff
@@ -71,325 +74,166 @@ public static function getUserName( Title $title, $namespaces, $fallbackToGlobal
-
-			// add "edit" item to "Leave a message" button
-			if ( $this->actionName == 'leavemessage' ) {
-				array_unshift( $actions, 'edit' );
-			}
-
-			foreach( $actions as $action ) {
-				if ( isset( $this->content_actions[$action] ) ) {
-					$actionMenu['dropdown'][$action] = $this->content_actions[$action];
-				}
-			}
-		}
-		$this->actionMenu = $actionMenu;
-
-		// don't show stats for user pages with too long titles (RT #68818)
-		if ( mb_strlen( $this->title ) > 35) {
-			$this->stats = false;
-		}
-
-		wfProfileOut( __METHOD__ );
-	}
-
-	/**
-	 * dirty way to get the user url
-	 * returns the full path of the users
-	 **/
-	public static function getUserURL() {
-		global $wgUser;
-
-		$user_name = $wgUser->mName;
-		$userURL = explode( '/', AvatarService::getUrl( $user_name ) );
-		$userURL = $userURL[count( $userURL ) -1];
-		return $userURL;
+		// render tabbed links below masthead
+		$this->tabs = $this->getTabs( $userName );
 	}
 
 	/**
 	 * Render header for blog post
 	 */
-	public function executeBlogPost() {
+	public function blogPost() {
 		wfProfileIn( __METHOD__ );
-		global $wgTitle, $wgLang, $wgOut;
-
 		// remove User_blog:xxx from title
-		$titleParts = explode( '/', $wgTitle->getText() );
+		$titleParts = explode( '/', $this->wg->Title->getText() );
 		array_shift( $titleParts );
 		$this->title = implode( '/', $titleParts );
 
 		// get user name to display in header
-		$this->userName = self::getUserName( $wgTitle, BodyController::getUserPagesNamespaces() );
+		$this->userName = self::getUserName( $this->wg->Title, BodyController::getUserPagesNamespaces() );
 
 		// render avatar (48x48)
 		$this->avatar = AvatarService::renderAvatar( $this->userName, 48 );
 
 		// link to user page
 		$this->userPage = AvatarService::getUrl( $this->userName );
 
-		if( $this->wg->EnableBlogArticles ) {
+		if ( $this->wg->EnableBlogArticles ) {
 			// link to user blog page
 			$this->userBlogPage = AvatarService::getUrl( $this->userName, NS_BLOG_ARTICLE );
 
 			// user blog page message
 			$this->userBlogPageMessage = wfMessage( 'user-blog-url-link', $this->userName )->inContentLanguage()->parse();
 		}
-		if( !empty( $this->wg->EnableGoogleAuthorInfo )
+		if ( !empty( $this->wg->EnableGoogleAuthorInfo )
 			&& !empty( $this->wg->GoogleAuthorLinks )
 			&& array_key_exists( $this->userName, $this->wg->GoogleAuthorLinks )
 		) {
 			$this->googleAuthorLink = $this->wg->GoogleAuthorLinks[$this->userName] . '?rel=author';
 		}
-		if( $this->app->wg->Request->getVal( 'action' ) == 'history' || $this->app->wg->Request->getCheck( 'diff' ) ) {
-			$this->navLinks = Wikia::link( $this->app->wg->title, wfMsg( 'oasis-page-header-back-to-article' ), array(), array(), 'known' );
-		}
-		// user stats (edit points, account creation date)
-		$this->stats = $this->getStats( $this->userName );
+
+		$this->showNavLinks = $this->wg->Request->getVal( 'action' ) == 'history' || $this->wg->Request->getCheck( 'diff' );
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "action" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)